threadloop/threadloop.py

```diff
@@ -90,8 +99,11 @@ def submit(self, fn, *args, **kwargs):
         :param kwargs: Kwargs to pass to coroutine
         :returns concurrent.futures.Future: future result of coroutine
         """
-        if not self._is_running:
-            raise ThreadNotStartedError()
+        if not self.is_ready():
+            raise ThreadNotStartedError(
+                "The thread has not been started yet, make sure start has "
+                "been called first like so: ThreadLoop().start()"
```

Comment:
Nit: If you've called submit, you already have a ThreadLoop instance. You don't need to say `ThreadLoop()` before starting that instance.